File: sandbox/backend/java/simple-eccomerce/src/main/java/com/simple/ecommerce/application/exception/CategoryNotFoundException.java
package com.simple.ecommerce.application.exception;

/**
 * Exception thrown when a category cannot be found.
 */
public class CategoryNotFoundException extends RuntimeException {
    
    public CategoryNotFoundException(String message) {
        super(message);
    }
    
    public CategoryNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }
    
    public CategoryNotFoundException(Long categoryId) {
        super("Category with ID " + categoryId + " not found");
    }
    
    /**
     * Creates a new exception for a category not found by name.
     *
     * @param categoryName the name of the category that wasn't found
     * @param exactMatch whether the search was for an exact match
     * @return the exception
     */
    public static CategoryNotFoundException forName(String categoryName, boolean exactMatch) {
        String message = "Category with name '" + categoryName + "' not found";
        if (exactMatch) {
            message += " (exact match)";
        }
        return new CategoryNotFoundException(message);
    }
} 

File: Projects/Testtrack/p911 - Simple eCommerce backend/application/exception/category_not_found_exception.py
from typing import Optional


class CategoryNotFoundException(Exception):
    def __init__(self, message: str, cause: Optional[Exception] = None):
        super().__init__(message)
        self.__cause__ = cause

    @classmethod
    def for_id(cls, category_id: int) -> 'CategoryNotFoundException':
        return cls(f"Category with ID {category_id} not found")

    @classmethod
    def for_name(cls, category_name: str, exact_match: bool = False) -> 'CategoryNotFoundException':
        message = f"Category with name '{category_name}' not found"
        if exact_match:
            message += " (exact match)"
        return cls(message)
